features/bootstrap/ElasticsearchContext.php

```diff
@@ -0,0 +1,128 @@
+    {
+        $this->client = $client;
+        $this->elasticsearchMappingsPath = $elasticsearchMappingsPath;
+        $this->elasticsearchFixturesPath = $elasticsearchFixturesPath;
+    }
+
+    /**
+     * @BeforeScenario @createIndexes
+     */
+    public function createIndexes(): void
+    {
+        $this->deleteIndexes();
+        $this->createIndexesAndMappings();
+        $this->loadFixtures();
+    }
+
+    /**
+     * @Given indexes and their mappings are created
+     */
+    public function thereAreIndexes(): void
+    {
+        $this->createIndexesAndMappings();
+    }
+
+    /**
+     * @Given indexes are deleted
+     */
+    public function thereAreNoIndexes(): void
+    {
+        $this->deleteIndexes();
+    }
+
+    /**
+     * @Given fixtures files are loaded
+     */
+    public function thereAreFixtures(): void
+    {
+        $this->loadFixtures();
+    }
+
+    private function createIndexesAndMappings(): void
+    {
+        $finder = new Finder();
+        $finder->files()->in($this->elasticsearchMappingsPath);
+
+        foreach ($finder as $file) {
+            $this->client->indices()->create([
+                'index' => $file->getBasename('.json'),
+                'body' => json_decode($file->getContents(), true),
+            ]);
+        }
+    }
+
+    private function deleteIndexes(): void
+    {
+        $finder = new Finder();
+        $finder->files()->in($this->elasticsearchMappingsPath)->name('*.json');
+
+        $indexClient = $this->client->indices();
+
+        foreach ($finder as $file) {
+            $index = $file->getBasename('.json');
+            if (!$indexClient->exists(['index' => $index])) {
+                continue;
+            }
+
+            $indexClient->delete(['index' => $index]);
+        }
+    }
+
+    private function loadFixtures(): void
+    {
+        $finder = new Finder();
+        $finder->files()->in($this->elasticsearchFixturesPath)->name('*.json');
+
+        foreach ($finder as $file) {
+            $index = $file->getBasename('.json');
+            $bulk = [];
+
+            foreach (json_decode($file->getContents(), true) as $document) {
+                if (null === $document['id'] ?? null) {
+                    $bulk[] = ['index' => ['_index' => $index, '_type' => DocumentMetadata::DEFAULT_TYPE]];
+                } else {
+                    $bulk[] = ['create' => ['_index' => $index, '_type' => DocumentMetadata::DEFAULT_TYPE, '_id' => (string) $document['id']]];
+                }
+
+                $bulk[] = $document;
+
+                if (0 === (\count($bulk) % 50)) {
+                    $this->client->bulk(['body' => $bulk]);
+                    $bulk = [];
+                }
+            }
+
+            if ($bulk) {
+                $this->client->bulk(['body' => $bulk]);
+            }
+        }
+
+        // Waiting for indexing...
+        sleep(5);
```

Comment:
I have to replace this by https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-refresh.html.